webapi/src/main/scala/org/knora/webapi/core/HttpServerZ.scala

```diff
@@ -30,19 +32,45 @@ object HttpServerZ {
         )
     } yield Middleware.cors(corsConfig)
 
+  private def metricsMiddleware() = {
+    // in order to avoid extensive amounts of labels, we should use the `pathLabelMapper`
+    // which maps paths containing slug values (`/project/iri/SOME-IRI`)
+    // to paths with a slug inserted instead (`/projects/iri/:iri`).
+    val slugs = Set(
+      "iri",
+      "shortcode",
+      "shortname"
+    )
+    @tailrec
+    def step(seg: List[Path.Segment], acc: Path): Path =
```

Comment:
The name of this function is very genetic, could you name it something like `replaceProjectSpecificPartsOfRoute`?

Reply:
I don't think `projectSpecific` should be in there, because the same will go for all other entities (Users, Resources, etc.). Also, I want to have it short, so the recursive calls don't look too bad (and it's an internal function so the visibility is small enough to warrant a short name. How do you feel about `replaceSlugs()`?

Reply:
Yes, that makes sense.`replaceSlugs()` sounds good to me.